pkg/config/configobserver/configobserver.go

```diff
@@ -262,64 +104,89 @@ func (c *Controller) ConfigChanged() (configCh <-chan struct{}, closeFn func())
 	}
 }
 
+// Fetches the token from cluster secret key
+func (c *Controller) fetchSecret(ctx context.Context, key string) (*v1.Secret, error) {
+	secret, err := c.kubeClient.CoreV1().Secrets("openshift-config").Get(ctx, key, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.V(4).Infof("%s secret does not exist", key)
+			err = nil
+		} else if errors.IsForbidden(err) {
+			klog.V(2).Infof("Operator does not have permission to check %s: %v", key, err)
+			err = nil
+		} else {
+			err = fmt.Errorf("could not check %s: %v", key, err)
+		}
+	}
+
+	return secret, err
+}
+
+// Updates the stored tokens from the secrets in the cluster. (if present)
+func (c *Controller) updateToken(ctx context.Context) error {
+	klog.V(2).Infof("Refreshing configuration from cluster pull secret")
+	secret, err := c.fetchSecret(ctx, "pull-secret")
+	if err != nil {
+		return err
+	}
+
+	var nextConfig config.Controller
+	if secret != nil {
+		var token string
+		token, err = tokenFromSecret(secret)
+		if err != nil {
+			return err
+		}
+		if len(token) > 0 {
+			nextConfig.Token = token
+			nextConfig.Report = true
```

Comment:
Hmm I am actually not sure about this line right now. Does it actually fix the original logic? It looks like it doesn't break anything right.... :) I am just wondering why [this](https://github.com/openshift/insights-operator/blob/master/pkg/config/configobserver/configobserver.go#L107) was set regardless the token.

Reply:
Well, I didn't notice any issues on my tests, but... maybe would be a good idea to keep the same logic.

Reply:
Well....it was (and maybe still is :-)) really confusing. I guess that originally it was overwritten at https://github.com/openshift/insights-operator/blob/master/pkg/config/configobserver/configobserver.go#L190 

It looks like it's not overwritten after your update so it's hopefully OK.